### PR TITLE
Addition of CSV

### PR DESCRIPTION
**Pull request by zekth**
Currently work in progress.
Ported tests from golang

**EDIT:**
Remaining quotes and CRLF tests to fix and we're done.

### CODE REVIEW

File: encoding/csv/test.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+import "./mod_test.ts";
```
Comment (ry) [inline encoding/csv/test.ts]: Why not `encoding/csv_test.ts` ?

File: encoding/csv/mod.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
+import { BufReader, BufState } from "../../io/bufio.ts";
```
Comment (ry) [inline encoding/csv/mod.ts]: Rename file to `encoding/csv.ts`

Please add 
```
// Ported from Go: Copyright 2009 The Go Authors.
// https://github.com/golang/go/blob/go1.12.5/src/encoding/csv/csv.go
// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
```

Comment (zekth) [reply encoding/csv/mod.ts]: Will surely do.

**Comment by zekth** [comment]
Remains weird cases with double quotes and multiline in fields to handle. For me it's landable now and tests are already here. So if you want to land it now i'm OK. There is a TODO comment added for remaining cases

File: encoding/csv.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  StartLine: number;
+  Line: number;
+  constructor(start: number, line: number, message: string) {
+    super(message);
+    this.StartLine = start;
+    this.Line = line;
+  }
+}
+
+export interface CsvParseOptions {
```
Comment (ry) [inline encoding/csv.ts]: Rename to `ParseOptions`

File: encoding/csv.ts [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  Comma: string;
+  Comment?: string;
+  TrimLeadingSpace: boolean;
+  LazyQuotes?: boolean;
+  FieldsPerRecord?: number;
```
Comment (ry) [inline encoding/csv.ts]: Fields should have a lower case first letter.

Comment (zekth) [reply encoding/csv.ts]: Is there a way to add an eslint rule for this? I think it could be a good addition.

**Pull request merged**